internal/cli/atlas/accesslogs/list.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslogs
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	success      = "success"
+	fail         = "fail"
+	listTemplate = `HOSTNAME	CLUSTER NAME	AUTH RESULT	LOG LINE {{range .AccessLogs}}
+{{if .Hostname}}{{.Hostname}} {{else}}N/A{{end}}{{.Hostname}}	{{if .ClusterName}}{{.ClusterName}} {{else}}N/A{{end}}	{{.AuthResult}}	{{.LogLine}}{{end}}
+`
+	missingClusterNameHostnameErrorMessage = "one between --%s and --%s must be set"
+	invalidValueAuthResultErrorMessage     = `--%s must be set to "%s" or "%s"`
+)
+
+type ListOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	hostname    string
+	clusterName string
+	start       string
+	end         string
+	nLogs       int
+	ipAddresses string
+	authResult  string
+	store       store.AccessLogsLister
+}
+
+func (opts *ListOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(store.AuthenticatedPreset(config.Default()))
+	return err
+}
+
+func (opts *ListOpts) Run() error {
+	if opts.clusterName != "" {
+		r, err := opts.store.AccessLogsByClusterName(opts.ConfigProjectID(), opts.clusterName, opts.newAccessLogOptions())
+		if err != nil {
+			return err
+		}
+
+		return opts.Print(r)
+	}
+
+	r, err := opts.store.AccessLogsByHostname(opts.ConfigProjectID(), opts.hostname, opts.newAccessLogOptions())
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *ListOpts) newAccessLogOptions() *atlas.AccessLogOptions {
+	var authResult *bool
+	if strings.EqualFold(opts.authResult, fail) {
+		t := false
+		authResult = &t
+	}
+
+	if strings.EqualFold(opts.authResult, success) {
+		t := true
+		authResult = &t
+	}
```

Comment:
[q] does something like this work?
```suggestion
	var authResult *bool

	if opts.authResult != "" {
		authResult = &strings.EqualFold(opts.authResult, success)
	}
```

Reply:
I don't see how the user can retrieve only the failed logs. Here we are only setting `authResult=true` and `authResult=nil`

Reply:
ohhh now I seee

Reply:
yes, it should work! Nice one!!! 💯 I am going to open a follow up PR!